quilt/tools/command.py

```diff
@@ -306,10 +306,44 @@ def push(package):
     pkgobj = PackageStore.find_package(owner, pkg)
     if pkgobj is None:
         raise CommandException("Package {owner}/{pkg} not found.".format(owner=owner, pkg=pkg))
+
+    obj_hashes = sorted(set(find_object_hashes(pkgobj.get_contents())))
+    total = len(obj_hashes)
+
+    headers = {
+        'Content-Encoding': 'gzip'
+    }
+
+    for idx, obj_hash in enumerate(obj_hashes):
+        print("Uploading %s (%d/%d)..." % (obj_hash, idx + 1, total))
+
+        response = session.get(
+            "{url}/api/blob/{owner}/{hash}".format(
+                url=QUILT_PKG_URL,
+                owner=owner,
+                hash=obj_hash
+            )
+        )
+        contents = response.json()
+
+        if not reupload:
+            s3_response = requests.head(contents['head'])
+            if s3_response.ok:
+                print("Fragment already uploaded; skipping.")
+                continue
```

Comment:
i guess python doesn't have named loops... very minor nit: this codeblock is long enough to where a comment would clarify what this `continue` applies to.